src/linker/Linker.Dataflow/DynamicallyAccessedMembersBinder.cs

```diff
@@ -0,0 +1,296 @@
+					yield return c;
+			}
+
+			if (memberTypes.HasFlag (DynamicallyAccessedMemberTypes.PublicConstructors)) {
+				foreach (var c in GetConstructorsOnType (typeDefinition, filter: null, bindingFlags: BindingFlags.Public))
+					yield return c;
+			}
+
+			if (memberTypes.HasFlag (DynamicallyAccessedMemberTypes.DefaultConstructor)) {
+				foreach (var c in GetConstructorsOnType (typeDefinition, filter: m => m.IsPublic && m.Parameters.Count == 0))
+					yield return c;
+			}
+
+			if (memberTypes.HasFlag (DynamicallyAccessedMemberTypes.NonPublicMethods)) {
+				foreach (var m in GetMethodsOnTypeHierarchy (typeDefinition, filter: null, bindingFlags: BindingFlags.NonPublic))
+					yield return m;
+			}
+
+			if (memberTypes.HasFlag (DynamicallyAccessedMemberTypes.PublicMethods)) {
+				foreach (var m in GetMethodsOnTypeHierarchy (typeDefinition, filter: null, bindingFlags: BindingFlags.Public))
+					yield return m;
+			}
+
+			if (memberTypes.HasFlag (DynamicallyAccessedMemberTypes.NonPublicFields)) {
+				foreach (var f in GetFieldsOnTypeHierarchy (typeDefinition, filter: null, bindingFlags: BindingFlags.NonPublic))
+					yield return f;
+			}
+
+			if (memberTypes.HasFlag (DynamicallyAccessedMemberTypes.PublicFields)) {
+				foreach (var f in GetFieldsOnTypeHierarchy (typeDefinition, filter: null, bindingFlags: BindingFlags.Public))
+					yield return f;
+			}
+
+			if (memberTypes.HasFlag (DynamicallyAccessedMemberTypes.NonPublicNestedTypes)) {
+				foreach (var t in GetNestedTypesOnType (typeDefinition, filter: null, bindingFlags: BindingFlags.NonPublic))
+					yield return t;
+			}
+
+			if (memberTypes.HasFlag (DynamicallyAccessedMemberTypes.PublicNestedTypes)) {
+				foreach (var t in GetNestedTypesOnType (typeDefinition, filter: null, bindingFlags: BindingFlags.Public))
+					yield return t;
+			}
+
+			if (memberTypes.HasFlag (DynamicallyAccessedMemberTypes.NonPublicProperties)) {
+				foreach (var p in GetPropertiesOnTypeHierarchy (typeDefinition, filter: null, bindingFlags: BindingFlags.NonPublic))
+					yield return p;
+			}
+
+			if (memberTypes.HasFlag (DynamicallyAccessedMemberTypes.PublicProperties)) {
+				foreach (var p in GetPropertiesOnTypeHierarchy (typeDefinition, filter: null, bindingFlags: BindingFlags.Public))
+					yield return p;
+			}
+
+			if (memberTypes.HasFlag (DynamicallyAccessedMemberTypes.NonPublicEvents)) {
+				foreach (var e in GetEventsOnTypeHierarchy (typeDefinition, filter: null, bindingFlags: BindingFlags.NonPublic))
+					yield return e;
+			}
+
+			if (memberTypes.HasFlag (DynamicallyAccessedMemberTypes.PublicEvents)) {
+				foreach (var e in GetEventsOnTypeHierarchy (typeDefinition, filter: null, bindingFlags: BindingFlags.Public))
+					yield return e;
+			}
+		}
+
+		public static IEnumerable<MethodDefinition> GetConstructorsOnType (TypeDefinition type, Func<MethodDefinition, bool> filter, BindingFlags? bindingFlags = null)
```

Comment:
These are local helpers, right? Why are they public?

Reply:
They are also used from ReflectionMethodBodyScanner - so maybe "internal" would be better.

Reply:
`DynamicallyAccessedMembersBinder` is already internal because our copy of `DynamicallyAccessedMemberTypes` is internal, so I think they are effectively internal already. Generally I prefer to use class-based accessibility modifiers for consistency with the rest of the codebase, especially since we have the ref assembly to control externally-visible public surface.